tests/unit/s2n_ktls_test_utils_test.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "utils/s2n_random.h"
+
+#define S2N_TEST_RECORD_TYPE 43
+
+/* Creates an iovec and sets it on new msghdr. This operation is verbose
+ * and not super interesting so its captured in a macro to make the tests
+ * easier to read. */
+#define INIT_MSGHDR(name, buf, len)       \
+    struct iovec name##_msg_iov = { 0 };  \
+    name##_msg_iov.iov_base = buf;        \
+    name##_msg_iov.iov_len = len;         \
+    struct msghdr name##_msg = { 0 };     \
+    name##_msg.msg_iov = &name##_msg_iov; \
+    name##_msg.msg_iovlen = 1;
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* test data */
+    uint8_t test_data[S2N_TLS_MAXIMUM_FRAGMENT_LENGTH] = { 0 };
+    struct s2n_blob test_data_blob = { 0 };
+    EXPECT_SUCCESS(s2n_blob_init(&test_data_blob, test_data, sizeof(test_data)));
+    EXPECT_OK(s2n_get_public_random_data(&test_data_blob));
+
+    /* Test the test. Tests the mock IO stuffer implementation */
+    for (size_t to_send = 1; to_send < S2N_TLS_MAXIMUM_FRAGMENT_LENGTH; to_send++) {
+        DEFER_CLEANUP(struct s2n_connection *server = s2n_connection_new(S2N_SERVER),
+                s2n_connection_ptr_free);
+        DEFER_CLEANUP(struct s2n_connection *client = s2n_connection_new(S2N_CLIENT),
+                s2n_connection_ptr_free);
+        DEFER_CLEANUP(struct s2n_test_ktls_io_pair io_pair = { 0 },
+                s2n_ktls_io_pair_free);
+        EXPECT_OK(s2n_test_init_ktls_stuffer_io(server, client, &io_pair));
+
+        /* Init send msghdr */
+        INIT_MSGHDR(send, test_data, to_send);
+        /* sendmsg */
+        ssize_t bytes_written = s2n_test_ktls_sendmsg_stuffer_io(server, &send_msg, S2N_TEST_RECORD_TYPE);
```

Comment:
Maybe it would make more sense if this function was called s2n_test_ktls_sendmsg()? Because that's what you're doing right? You're mocking sendmsg functionality (I think).

The reason why I'm suggesting it this is the current name doesn't give me any idea what the function is doing.

Reply:
I wanted to capture the sendmsg using stuffer_io aspect of this impementation. Maybe `s2n_test_ktls_sendmsg_buffer_io` or `s2n_test_ktls_sendmsg_using_buffer_io`?

So there are going to be other impl that do different things: [s2n_test_ktls_sendmsg_validate](https://github.com/aws/s2n-tls/blob/97fb757b5c8bfe6f4c9c835e0e76f34d9f195f32/tests/testlib/s2n_ktls_test_utils.c#L200) and [s2n_test_ktls_sendmsg_err](https://github.com/aws/s2n-tls/blob/97fb757b5c8bfe6f4c9c835e0e76f34d9f195f32/tests/testlib/s2n_ktls_test_utils.c#L232C9-L232C34)